content/docs/visualization/template_reference.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Template reference
+sort_rank: 6
+---
+
+# Template reference
+
+Prometheus has support for templating in both the summary/description field of
+alerts and also exposed as html endpoints. Templates have the ability to run
+queries against the databases, iterate over data, use conditionals, format data
+etc. This is based on the [go templating](http://golang.org/pkg/text/template/)
+system.
+
+## Data Structures
+
+The primary structure is the sample, defined as:
+
+```
+type sample struct {
+        Labels map[string]string
+        Value  float64
+}
+```
+
+The `__name__` label is the name of the metric.
+
+`[]sample` means a list of samples.
+
+## Functions
+In addition to the [default
+functions](http://golang.org/pkg/text/template/#hdr-Functions) provided by go
+templating, prometheus provides functions to make writing templates easier.
```

Comment:
s/prometheus/Prometheus/

"...provides functions for easier processing of query results in templates."